validator/src/main.rs

```diff
@@ -271,70 +271,58 @@ fn get_rpc_node(
-        .get_account(vote_pubkey)
-        .map_err(|err| format!("Vote account not found: {}", err.to_string()))?;
+    let vote_account = rpc_client
+        .get_account(vote_account_address)
+        .map_err(|err| format!("vote account not found: {}", err.to_string()))?;
 
-    if found_vote_account.owner != solana_vote_program::id() {
+    if vote_account.owner != solana_vote_program::id() {
         return Err(format!(
             "not a vote account (owned by {}): {}",
-            found_vote_account.owner, vote_pubkey
+            vote_account.owner, vote_account_address
         ));
     }
 
-    let found_node_account = rpc_client
-        .get_account(node_pubkey)
+    let identity_account = rpc_client
+        .get_account(identity_pubkey)
         .map_err(|err| format!("Identity account not found: {}", err.to_string()))?;
 
-    let found_vote_account = solana_vote_program::vote_state::VoteState::from(&found_vote_account);
-    if let Some(found_vote_account) = found_vote_account {
-        if found_vote_account.authorized_voters().is_empty() {
+    let vote_state = solana_vote_program::vote_state::VoteState::from(&vote_account);
+    if let Some(vote_state) = vote_state {
+        if vote_state.authorized_voters().is_empty() {
             return Err("Vote account not yet initialized".to_string());
         }
 
-        let epoch_info = rpc_client
-            .get_epoch_info()
-            .map_err(|err| format!("Failed to get epoch info: {}", err.to_string()))?;
-
-        let mut authorized_voter;
-        authorized_voter = found_vote_account.get_authorized_voter(epoch_info.epoch);
-        if authorized_voter.is_none() {
-            // Must have gotten a clock on the boundary
-            authorized_voter = found_vote_account.get_authorized_voter(epoch_info.epoch + 1);
-        }
-
-        let authorized_voter = authorized_voter.expect(
-            "Could not get the authorized voter, which only happens if the
-            client gets an epoch earlier than the current epoch,
-            but the received epoch should not be off by more than
-            one epoch",
-        );
-
-        if authorized_voter != *voting_pubkey {
+        if vote_state.node_pubkey != *identity_pubkey {
             return Err(format!(
-                "account's authorized voter ({}) does not match to the given voting keypair ({}).",
-                authorized_voter, voting_pubkey
+                "vote account's identity ({}) does not match the validator's identity {}).",
+                vote_state.node_pubkey, identity_pubkey
             ));
         }
-        if found_vote_account.node_pubkey != *node_pubkey {
-            return Err(format!(
-                "account's node pubkey ({}) does not match to the given identity keypair ({}).",
-                found_vote_account.node_pubkey, node_pubkey
-            ));
+
+        for (_, vote_account_authorized_voter_pubkey) in vote_state.authorized_voters().iter() {
```

Comment:
Make sure the user supplied all the `--authorized-voter` keypairs that the vote account currently knows about.